### DISCUSSION TITLE
Generation of synthetic segmented data

### DISCUSSION THREAD

**Discussion by Louisducamp** [mobaidoctor/med-ddpm#27] [Q&A]
Bonjour,

First of all : Thank you for your amazing work and the paper ! 

In the context of my master thesis relying on anonymization of head volume, I'm willing to generate a Dicoms image and its Rtstructs.

I basically have in my dataset a niifti image cropped as 128x128x128 around the head and a niftii mask with mutliple labels as for you : 

- 1 for the body
- 2 for the brainsteem
- 3 for the right eye
- etc

So I wanna finetune your model 128 to get faster. 

My question is : Do i need to change anything in the command line or dataset.py for the mask attention because I'm not using the same structure as you (1 for head, 2 for tumor) 

Also do you have other pretrained generation model for the general one ? Such as different dimensions ? 

Thank you in advance for your answer

Louis

**Comment by Louisducamp** [comment]
Here is the code I modified so 

in dtypes.py : 

```
class LabelEnum(IntEnum):
    BACKGROUND = 0
    BODY_EXTREMITIES = 1
    OPTICNERVE_R = 2
    LACRIMAL_R = 3
    BRAINSTEM = 4
    CHIASM = 5
    LACRIMAL_L = 6
    COCHLEA_L = 7
    LENS_L = 8
    EYE_L = 9
    COCHLEA_R = 10
    EYE_R = 11
    LENS_R = 12
    CEREBELLUM = 13
```
and 
```
class NiftiPairImageGenerator(Dataset):
    def __init__(self,
            input_folder: str,
            target_folder: str,
            input_size: int,
            depth_size: int,
            input_channel: int = 14,
            transform=None,
            target_transform=None,
            full_channel_mask=False,
            combine_output=False
        ):
```

and in dataset.py
```

    def label2masks(self, masked_img):
        result_img = np.zeros(masked_img.shape + ( self.input_channel - 1,))
        result_img[masked_img==LabelEnum.BODY_EXTREMITIES.value, 0] = 1
        result_img[masked_img==LabelEnum.OPTICNERVE_R.value, 1] = 1
        result_img[masked_img==LabelEnum.LACRIMAL_R.value, 2] = 1
        result_img[masked_img==LabelEnum.BRAINSTEM.value, 3] = 1
        result_img[masked_img==LabelEnum.CHIASM.value, 4] = 1
        result_img[masked_img==LabelEnum.LACRIMAL_L.value, 5] = 1
        result_img[masked_img==LabelEnum.COCHLEA_L.value, 6] = 1
        result_img[masked_img==LabelEnum.LENS_L.value, 7] = 1
        result_img[masked_img==LabelEnum.EYE_L.value, 8] = 1
        result_img[masked_img==LabelEnum.COCHLEA_R.value, 9] = 1
        result_img[masked_img==LabelEnum.EYE_R.value, 10] = 1
        result_img[masked_img==LabelEnum.LENS_R.value, 11] = 1
        result_img[masked_img==LabelEnum.CEREBELLUM.value, 12] = 1
        return result_img
```

and what I need to change in train.py is `parser.add_argument('--num_class_labels', type=int, default=3)` ?
